### PR TITLE
Issue #2988112 by chr.fritsch: Integrate select2 module

### PR DESCRIPTION
**Pull request by chrfritsch**
Make sure these boxes are checked before submitting your pull request - thank you!

- [ ] All coding styles are fulfilled. ([How to check for cs issues?](https://github.com/BurdaMagazinOrg/thunder-dev-tools/blob/master/README.md#code-style-guidelines))
- [ ] All tests are running locally. ([How to run the test?](https://github.com/BurdaMagazinOrg/thunder-distribution/blob/develop/docs/development.md#how-to-run-the-tests))
- [ ] Necessary update hooks are provided.
- [ ] User roles have correct access for new introduced permission.
- [ ] Every thunder module has a README.md in its root. Follow [this guidelines](https://www.drupal.org/node/2181737), but we don't need every topic.
- [ ] Code is covered with well-balanced amount of inline comments.

If you are really awesome, then your feature is covered by additional tests. Well done!

### CODE REVIEW

File: thunder.install [hunk 1]
```diff
@@ -951,3 +951,26 @@ function thunder_update_8120() {
...
   return $thunderUpdater->logger()->output();
 }
+
+/**
+ * Enable select2boxes module.
+ */
+function thunder_update_8121() {
+  /** @var \Drupal\thunder_updater\Updater $thunderUpdater */
+  $thunderUpdater = \Drupal::service('thunder_updater');
+  if ($thunderUpdater->installModules(['thunder__thunder_update_8121' => 'select2boxes'])) {
```
Comment (mtodor) [inline thunder.install]: If module install fails, then checklist entry should be also marked as failed.

File: thunder.install [hunk 2]
```diff
@@ -951,3 +951,26 @@ function thunder_update_8120() {
...
+    /** @var \Drupal\thunder_updater\Updater $thunderUpdater */
+    $thunderUpdater = \Drupal::service('thunder_updater');
```
Comment (mtodor) [inline thunder.install]: There is already ` $thunderUpdater` is already assigned - line 960.

File: modules/thunder_updater/updates.yml [hunk 3]
```diff
@@ -209,3 +209,11 @@ v2.16:
         '#description_successful': '<p>The installation was successful.</p>'
         '#description_failed': '<p><strong>Installation failed</strong><p>'
         thunder_article_config: { '#text': 'Move scheduler link back from local tasks to menu tab', '#url': 'internal:/admin/config/thunder_article/configuration' }
+v2.19:
+    '#title': 'Update to Thunder 2.19'
+    '#description': '<h3>These are the new features for this release:</h3>'
+    thunder__thunder_update_8121:
+        '#title': 'Enable "Select2Boxes" module.'
+        '#description': "<p>Provide Select2 field widgets for forms and views filters.</p>"
+        '#description_failed': '<p><strong>Installation failed</strong>, visit the extension page and try to enable the module manually.</p>'
```
Comment (mtodor) [inline modules/thunder_updater/updates.yml]: Should we also add that tags and channels should be changed to use select2?

**Review by dbosen — APPROVED** [review]
Juhu!

**Pull request merged**